pkg/state/state_skel.go

```diff
@@ -214,27 +215,39 @@ func (s *stateSkel) createOrUpdateObjs(
 
 		s.addStateSpecificLabels(desiredObj)
 
-		err := s.createObj(ctx, desiredObj)
-		if err == nil {
-			// object created successfully
-			continue
-		}
-		if !k8serrors.IsAlreadyExists(err) {
-			// Some error occurred
+		desiredRev, err := revision.CalculateRevision(desiredObj)
+		if err != nil {
 			return err
 		}
+		revision.SetRevision(desiredObj, desiredRev)
```

Comment:
nit: you only need to set this if an update is required.

that said, im fine with either keeping it here or moving it to after L247

Reply:
We also need to set revision when we create the object at line 236.